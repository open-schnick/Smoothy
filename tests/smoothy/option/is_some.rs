use crate::failing_assertion;
use smoothy::{assert_that, BasicAsserter, EqualityAssertion, OptionAssertion};

mod assert_option {
    use super::*;

    #[test]
    fn succeeds() {
        let option: Option<String> = Some(String::new());

        assert_that(option).is_some();
    }

    #[test]
    fn fails() {
        failing_assertion!({
            let option: Option<()> = None;

            assert_that(option).is_some();
        });
    }
}

mod assert_some {
    use super::*;

    #[test]
    fn transforms_to_basic_asserter() {
        let option: Option<String> = Some(String::new());

        let asserter: BasicAsserter<String> = assert_that(option).is_some().and_value();
        asserter.equals("");
    }
}
