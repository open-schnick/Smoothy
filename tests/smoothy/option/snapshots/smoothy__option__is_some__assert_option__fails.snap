---
source: tests/smoothy/option/is_some.rs
expression: assertion_failed_output
---
[31mAssertion failed![0m

Expected
  [31mNone[0m
to be Some
